data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,498 @@
+from constants.cdr_cleaner import clean_cdr as clean_consts
+from sandbox import check_and_create_sandbox_dataset
+from constants import bq_utils as bq_consts
+
+LOGGER = logging.getLogger(__name__)
+LOGS_PATH = 'LOGS'
+
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+TABLE_INFORMATION_SCHEMA = jinja_env.from_string("""
+SELECT *
+FROM `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS`
+""")
+DEACTIVATED_PIDS_QUERY = jinja_env.from_string("""
+SELECT DISTINCT *
+FROM `{{project}}.{{dataset}}.{{table}}`
+""")
+RESEARCH_ID_QUERY = jinja_env.from_string("""
+SELECT DISTINCT research_id
+FROM `{{project}}.{{prefix_regex}}_combined._deid_map`
+WHERE person_id = {{pid}}
+""")
+CHECK_PID_EXIST_QUERY = jinja_env.from_string("""
+SELECT
+COALESCE(COUNT(*), 0) AS count
+FROM `{{project}}.{{dataset}}.{{table}}`
+WHERE person_id = {{pid}}
+""")
+# Queries to create tables in associated sandbox with rows that will be removed per cleaning rule, two different queries
+# for tables with standard entry dates vs. tables with start and end dates
+SANDBOX_QUERY_DATE = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` AS (
+SELECT *
+FROM `{{project}}.{{dataset}}.{{table}}`
+WHERE person_id = {pid}
+AND {{date_column}} >= (SELECT deactivated_date
+FROM `{{deactivated_pids_project}}.{{deactivated_pids_dataset}}.{{deactivated_pids_table}}`
+WHERE person_id = {{pid}}))
+""")
+SANDBOX_QUERY_END_DATE = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` AS (
+SELECT *
+FROM `{{project}}.{{dataset}}.{{table}}`
+WHERE person_id = {{pid}}
+AND (CASE WHEN {{end_date_column}} IS NOT NULL THEN {{end_date_column}} >= (SELECT deactivated_date
+FROM `{{deactivated_pids_project}}.{{deactivated_pids_dataset}}.{{deactivated_pids_table}}`
+WHERE person_id = {{pid}} ) ELSE CASE WHEN {{end_date_column}} IS NULL THEN {{start_date_column}} >= (
+SELECT deactivated_date
+FROM `{{deactivated_pids_project}}.{{deactivated_pids_dataset}}.{{deactivated_pids_table}}`
+WHERE person_id = {{pid}}) END END))
+""")
+# Queries to truncate existing tables to remove deactivated EHR PIDS, two different queries for
+# tables with standard entry dates vs. tables with start and end dates
+CLEAN_QUERY_DATE = jinja_env.from_string("""
+SELECT *
+FROM `{{project}}.{{dataset}}.{{table}}`
+WHERE person_id != {{pid}}
+AND {{date_column}} < (SELECT deactivated_date
+FROM `{{deactivated_pids_project}}.{{deactivated_pids_dataset}}.{{deactivated_pids_table}}`
+WHERE person_id = {{pid}})
+""")
+CLEAN_QUERY_END_DATE = jinja_env.from_string("""
```

Comment:
nit:  i think most pycharm users like the double blank lines between high level module functions and statements.